### PR TITLE
Optimize requests

### PR DESCRIPTION
**Pull request by MrMeison**
Fix to get Bing Maps tiles

* Add metadata server
* Update packages
* Remove unused code
* Optimize rendering
* Use Promise-based async operations
* Use static markup for control buttons group
* Update API version to  1.6.0

### CODE REVIEW

File: src/globemap.ts [hunk 1]
```diff
@@ -24,51 +24,44 @@
...
+
+    // powerbi.extensibility.utils.type
+    import ValueType = powerbi.extensibility.utils.type.ValueType;
 
     interface ExtendedPromise<T> extends IPromise<T> {
         always(value: any): void;
     }
-
     export class GlobeMap implements IVisual {
+        LocalStorageService: IStorageService;
```
Comment (ignatvilesov) [inline src/globemap.ts]: I suppose that it should be in lowercase and should be private.

File: src/globemap.ts [hunk 2]
```diff
@@ -472,33 +518,147 @@ module powerbi.extensibility.visual {
...
+         * @private
+         * @param {number} level map lavel
+         * @param {string} urlTemplate url template
+         * @example https://ecn.{subdomain}.tiles.virtualearth.net/tiles/r{quadkey}.jpeg?g=5691&mkt={culture}&shading=hill
+         * @param {string[]} subdomains list of subdomauns
+         * @returns {{ [quadKey: string]: string }} Object <quadKey> : <image url>
+         * @memberOf GlobeMap
+         */
+        private generateQuadsByLevel(level: number, urlTemplate: string, subdomains: string[]): TileMap {
+            const COUNT_SERVERS = subdomains.length;
```
Comment (ignatvilesov) [inline src/globemap.ts]: Please specify a type of the variable.
It's a local variable, I suppose that it should be in pascal case.

File: src/globemap.ts [hunk 3]
```diff
@@ -472,33 +518,147 @@ module powerbi.extensibility.visual {
...
+            urlTemplate = urlTemplate.replace("{culture}", this.currentLanguage);
+            const result: { [quadKey: string]: string } = {};
+            const COUNT_TILES: number = 4;
```
Comment (ignatvilesov) [inline src/globemap.ts]: Please specify a type of the variable.
It's a local variable, I suppose that it should be in pascal case.

File: src/globemap.ts [hunk 4]
```diff
@@ -472,33 +518,147 @@ module powerbi.extensibility.visual {
...
+            return texture;
+        }
+
+        /**
+         * Load tiles of Bing Maps
+         * @param jCanvas jQuery convas object
+         * @param tiles map of tiles
+         * @param successCallback call this function when all tiles of the map are successfully loaded
+         */
+        private loadTiles(canvasEl: HTMLCanvasElement, tiles: TileMap, successCallback: Function) {
```
Comment (ignatvilesov) [inline src/globemap.ts]: Please specify a return type of the method.

File: src/dataInterfaces.ts [hunk 5]
```diff
@@ -52,6 +55,34 @@ module powerbi.extensibility.visual {
...
+
+    export interface TileMap {
+        [quadKey: string]: string ;
+    }
+
+    export interface CanvasCoordinate {
+        x: number;
+        y: number;
+
+
```
Comment (ignatvilesov) [inline src/dataInterfaces.ts]: Please remove this empty line.

File: src/dataInterfaces.ts [hunk 6]
```diff
@@ -52,6 +55,34 @@ module powerbi.extensibility.visual {
...
+    }
+
+    export interface TileMap {
+        [quadKey: string]: string ;
+    }
+
+    export interface CanvasCoordinate {
+        x: number;
+        y: number;
+
```
Comment (ignatvilesov) [inline src/dataInterfaces.ts]: Please remove this empty line.

File: src/settings.ts [hunk 7]
```diff
@@ -25,13 +25,13 @@
...
 
 module powerbi.extensibility.visual {
+    // powerbi.extensibility.utils.dataview
     import DataViewObjectsParser = powerbi.extensibility.utils.dataview.DataViewObjectsParser;
 
     export class GlobeMapSettings extends DataViewObjectsParser {
         public dataPoint: DataPointSettings = new DataPointSettings();
     }
 
     export class DataPointSettings {
```
Comment (ignatvilesov) [inline src/settings.ts]: Does anything use it?

Comment (MrMeison) [inline src/settings.ts]: Yes, but we will can to add new settings later and it's easier with separate settings file

**Pull request merged**